gcovr/gcov.py

```diff
@@ -375,6 +375,37 @@ def parse_tag_line(
             return True
 
         if line.startswith('function '):
+            # function tags look like:
+            #   function X called 0 returned 0% blocks executed 0%
+            #   function X Y called 0 returned 0% blocks executed 0%
+            #   function X Y Z called 0 returned 0% blocks executed 0%
+            #   function X Y Z ... called 0 returned 0% blocks executed 0%
+            # where the percentage should usually be a count.
+            # we could also use a regex: "^function\s+(.+)\s+called\s+(\d+)\s+"
```

Comment:
Indeed a regex would be more elegant. For normal source-lines we're avoiding regexes due to performance concerns, but TBH this hasn't been benchmarked. Since function tags are comparatively rare, it's OK if the parsing code is less efficient.

Reply:
Currently the implementation is consistent with branch coverage. For sure it could be done with a regex, but won't be part of this PR